commonv5/ts/analyze-task.ts

```diff
@@ -1,12 +1,19 @@
+import * as path from "path";
 import * as tl from "azure-pipelines-task-lib/task";
 import Scanner, { ScannerMode } from "./sonarqube/Scanner";
 import JavaVersionResolver from "./helpers/java-version-resolver";
+import { extractAndPublishTaskVersion } from "./helpers/task-version-utils";
 import { EndpointType, EndpointData } from "./sonarqube/Endpoint";
 import { sanitizeVariable, PROP_NAMES } from "./helpers/utils";
 
 const JAVA_11_PATH_ENV_NAME = "JAVA_HOME_11_X64";
 
 export default async function analyzeTask(rootPath: string, isSonarCloud: boolean = false) {
+  if (tl.getVariable("IT_TRIGGER_BUILD")) {
```

Comment:
I'm not clear what's happening here. It looks like there is test-specific behaviour hard-coded into the "real" tasks to make them export extra data if they detect the ITs are being run. The same data is then passed to the ITs, which check the values are the same.

What is the overall purpose of these checks?
Is it to check that ITs are testing the correct tasks i.e. that the new tasks have been correctly published and installed in the test org?

Or is to check that the published tasks have the expected version numbers? If so, isn't the data just being extracted from the same `.json` files in two different places, once in the product tasks and once in the yaml file? Since the ultimate source of the data is the same in each case, I'm don't understand how they would ever be different.

Reply:
I wanted to make sure that the extension i published during the QA is the same those pipeline/projects used to run, i don't like also the way it is but i haven't found other way to do so.

Reply:
Can we detect if the unpublish/republish steps fail? That might be enough.

Reply:
All steps are conditioned to succeeded previous step, is it enough then ?

Reply:
I would have thought that would be enough; if the unpublish/publish works, we should be able to assume that the correct version is installed.

Reply:
Yes, thanks, however i had a thought yesterday and this morning that this mechanism shall not work if multiple PRs are opened in the same time, because the same version (or a newer) can be installed and replace the one that a PR has installed itself, for the moment since i'm working alone on that it should be ok, but will not be sustanaible and will have to think of being possible to have multiple extension (with revision on the semver and possibly a different id) installed at the same time on one publisher.

Reply:
Good point. Maybe create a ticket to address that later?

Reply:
Yep !

Reply:
Ticket created : https://jira.sonarsource.com/browse/VSTS-276